docs/architecture/adr-050-improved-trusted-peering.md

```diff
@@ -0,0 +1,59 @@
+# ADR 50: Improved Trusted Peering
+
+## Changelog
+* 22-10-2019: Initial draft
+
+## Context
+
+When `max_num_inbound_peers` or `max_num_outbound_peers` of a node reached, the node cannot spare more slots to any peer 
+by inbound or outbound. Therefore, after certain period of disconnection, any important peering can be lost indefinitely 
+because all slots consumed by other peers, and the node stop trying to dial the peer anymore.
+
+This is happening because of two reasons, exponential backoff and absence of unconditional peering feature for trusted peers.
+
+
+## Decision
+
+We would like to suggest solving this problem by introducing two parameters in `config.toml`, `wildcard_peer_ids` and 
+`maximum_dial_period`. 
+
+1) `wildcard_peer_ids`
+
+A node operator inputs list of ids of peers which are allowed to be connected by both inbound or outbound regardless of 
+`max_num_inbound_peers` or `max_num_outbound_peers` of user's node reached or not.
+
+2) `maximum_dial_period`
+
+Terms between each dial to each persistent peer will not exceed `maximum_dial_period` during exponential backoff. 
```

Comment:
Maybe it's enough to extend the max from 24h to 48h? and lower the exponent?

Reply:
For validators, monitoring all connection everyday is not very trivial operation. From our experience, running tens of networks sometimes result in missing to alert relatively crucial connection for days. 

For persistent peers, we feel backoff until 30minute is more than enough, from our experience. Never want it to back off 24 or 48 hours.